modules/extensions/src/api/exec.ts

```diff
@@ -0,0 +1,59 @@
+import { extensionPort, proxy } from "../util/comlink";
+import { Env, ExecOptions, ExecResult } from "src/types";
+
+interface ExecArgs {
+  splitStderr?: boolean;
+  env?: Env;
+  args: Array<string> | string;
+  onOutput?: (output: string) => void;
+  onStdErr?: (stderr: string) => void;
+  onError?: (error: Error) => void;
+  onEnd?: (
+    res: ExecResult & {
+      output: string;
+    }
+  ) => void;
+}
+
+export default async function exec({
+  splitStderr,
+  args,
+  env,
+  onOutput = () => {},
+  onStdErr = () => {},
+  onError = () => {},
+  onEnd = () => {},
```

Comment:
i dont think we need onEnd, the promise itself should signal the end, no?

Reply:
The exec function returns a disposer while using a callback to signal for when it ends.  If we wait until the command finishes, there will be no way to stop a running process.